tuning/sft_trainer.py

```diff
@@ -218,15 +219,22 @@ def train(
             # TODO: Fix this, currently unreachable due to crashing in batch encoding tokenization
             # We should do this validation up front, then do the encoding, then handle the collator
             raise ValueError("Response template is None, needs to be set for training")
-        if data_args.dataset_text_field is None:
-            raise ValueError("Dataset_text_field is None, needs to be set for training")
         data_collator = DataCollatorForCompletionOnlyLM(
             response_template_ids,
             tokenizer=tokenizer,
             ignore_index=configs.IGNORE_INDEX,
         )
         packing = False
 
+    # Currently we support formatted datasets with single sequence instances.
+    if (data_args.dataset_text_field is None) and (data_args.data_formatter_template is None):
```

Comment:
I noticed that in this condition, it explicitly checks `None`, and the next it's checking for `None` or empty strings. Any thoughts on changing this to `if data_args.dataset_text_field and data_args.data_formatter_template :` so that `None` and empty string are treated consistently here, since I think that makes sense for these checks?

Reply:
updated but first check is checking that at least one is set , second check is checking that only one is set. we could consolidate to one check in one if statement , but just making error messages more specific in 2 separate checks